src/deliverable.rs

```diff
@@ -0,0 +1,14 @@
+use std::sync::mpsc;
+
+use transaction::DeliveryResult;
+
+/// The trait that handles the completion of the transaction (delivery result)
+pub trait Deliverable : Send + 'static {
+    fn complete(self, result: DeliveryResult);
+}
+
+impl Deliverable for mpsc::Sender<DeliveryResult> {
+    fn complete(self, result: DeliveryResult) {
+        let _ = self.send(result);
```

Comment:
Good point, yeah it's unimportant because it's only used for the tests. I moved it over to the test mod, we don't really need to expose the `Deliverable` impl for `mpsc::Sender` anyways..